tests/garage/envs/bullet/test_bullet_env.py

```diff
@@ -0,0 +1,90 @@
+"""Test module for BulletEnv"""
+import pickle
+
+import gym
+import pybullet_envs
+from pybullet_utils.bullet_client import BulletClient
+import pytest
+
+from garage.envs.bullet import BulletEnv
+from tests.helpers import step_env
+
+
+@pytest.mark.parametrize('env_ids', [pybullet_envs.getList()])
+def test_can_step(env_ids):
+    """Test Bullet environments can step"""
+
+    for env_id in env_ids:
+        # extract id string
+        env_id = env_id.replace('- ', '')
+        if env_id == 'KukaCamBulletEnv-v0':
+            # Kuka environments calls py_bullet.resetSimulation() in reset()
+            # unconditionally, which globally resets other simulations. So
+            # only one Kuka environment is tested.
+            continue
```

Comment:
why does the reset of `KukaCamBulletEnv-v0` matter? every iteration creates a new bullet environment, so the reset in the last iteration should not impact the next iteration. Or am I missing something?

Reply:
I agree. Honestly I don't know why this is failing.
 
If two Kuka environments are tested, the second Kuka environment will complain about "py_bullet not connected to server" in its `reset()`. 

Some thoughts:
Kuka environment implementation doesn't store client-id, thus it assumes one running environment at a time. If there are two parallel Kuka environments, calling reset/disconnect will affect the other one. 

Is it possible that the first environment is garbage collected while the second one is running?  

My plan was to introduce bullet first. If we plan to use Kuka environment in a parallel scenario, I will spend more time and try to understand this.

Reply:
My theory: `pybullet.resetSimulation()` closes both the client and the server, while `client.resetSimulation()` closes only the client. 

Please go ahead with this broken `KukaCamBulletEnv-v0` and fix it later. 

If multiple instances are problematic, this env cannot be used for training, since the sampler will create multiple instances. We should probably document this behaviour somewhere, or even disable `KukaCamBulletEnv-v0` until the problem is fixed.